Objects/abstract.c

```diff
@@ -495,6 +494,47 @@ _Py_add_one_to_index_C(int nd, Py_ssize_t *index, const Py_ssize_t *shape)
     }
 }
 
+int
+PyBuffer_SizeFromFormat(const char *format)
+{
+    PyObject *structmodule;
+    PyObject *Struct = NULL;
+    PyObject *structobj = NULL;
+    PyObject *fmt = NULL;
+    int itemsize;
```

Comment:
initialize itemsize to -1, so the error label can be removed, and you can just use "goto done".